pymodbus/datastore/context.py

```diff
@@ -27,6 +31,43 @@ def decode(self, fx):
         """
         return self._fx_mapper[fx]
 
+    async def async_getValues(self, fc_as_hex: int, address: int, count: int = 1) -> list[int | bool] | ExceptionResponse:
+        """Get `count` values from datastore.
+
+        :param fc_as_hex: The function we are working with
+        :param address: The starting address
+        :param count: The number of values to retrieve
+        :returns: The requested values from a:a+c
+        """
+        return self.getValues(fc_as_hex, address, count)
+
+    async def async_setValues(self, fc_as_hex: int, address: int, values: list[int]) -> None | ExceptionResponse:
```

Comment:
setValues return None

Remember "raise" is not a return.

Reply:
An ExceptionResponse might be generated at PDU level, the datastore is only concerned about values.

Reply:
It might be the idea, but setValues will return response https://github.com/pymodbus-dev/pymodbus/blob/dev/pymodbus/datastore/remote.py#L60 if it is error one, I don't have opinion if it should raise it instead, but that is current implementation.

Ofcourse the baseclass for those responses might not be ExceptionResponse, that can be fixed.

Reply:
Again, remote is off and quite buggy. setValues should not return anything, but might raise and exception.

If you start testing remote, you will find a lot of problems....most of them based in the fact that the current datastore is not async.

Reply:
great, I updated the typehint for setValues